src/webviews/components/issue/common/editor/Editor.tsx

```
@@ -0,0 +1,289 @@
+        strong: { open: '*', close: '*', mixable: true, expelEnclosingWhitespace: true },
+    },
+);
+
+const mdParser = new MarkdownParser(schema, defaultMarkdownParser.tokenizer, {
+    ...defaultMarkdownParser.tokens,
+    mention: {
+        node: 'mention',
+    },
+});
+
+/**
+ * IMPORTANT: outer div's "suggestion-item-list" class is mandatory. The plugin uses this class for querying.
+ * IMPORTANT: inner div's "suggestion-item" class is mandatory too for the same reasons
+ */
+const getMentionSuggestionsHTML = (items: any[]) =>
+    '<div class="suggestion-item-list">' +
+    items.map((i) => `<div class="suggestion-item">${i.name}</div>`).join('') +
+    '</div>';
+
+const reactPropsKey = new PluginKey('reactProps');
+
+function reactProps(initialProps: any) {
+    return new Plugin({
+        key: reactPropsKey,
+        state: {
+            init: () => initialProps,
+            apply: (tr, prev) => tr.getMeta(reactPropsKey) || prev,
+        },
+    });
+}
+
+interface UserType {
+    displayName: string;
+    mention: string;
+    emailAddress?: string;
+    avatarUrl?: string;
+}
+
+export function useEditor<T extends UserType>(props: {
+    value: string;
+    enabled: boolean;
+    onSave?: (input: string) => void;
+    onChange: (input: string) => void;
+    fetchUsers?: (input: string) => Promise<T[]>;
+}) {
+    const viewHost = useRef<HTMLDivElement>(null);
+    const view = useRef<EditorView | null>(null);
+    const [content, setContent] = useState(props.value || '');
+
+    // Prevents unnecessary calls to fetchUsers
+    const debouncedFetch = props.fetchUsers && debounce(props.fetchUsers, 3000, { leading: true, maxWait: 1 });
+
+    const handleSave = useCallback(() => {
+        const mdContent: string = props.enabled ? mdSerializer.serialize(view.current!.state.doc) : content;
+
+        if (mdContent.trim().length > 0) {
+            try {
+                props.onSave?.(mdContent);
+            } catch {
```

what is this code saving? what's the implication of failing to save? should we throw an error here?

---

I took this logic from the BB implementation, this doesn't seem right. Will change!